qcodes/utils/deprecate.py

```diff
@@ -3,6 +3,21 @@
 from typing import Optional, Callable
 
 
+def issue_deprecation_warning(
+    what: str,
+    reason: Optional[str] = None,
+    alternative: Optional[str] = None
+) -> None:
+    msg = f'The {what} is deprecated'
+    if reason is not None:
+        msg += f', because {reason}'
+    msg += '.'
+    if alternative is not None:
+        msg += f' Use \"{alternative}\" as an alternative.'
+
+    warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
I as surprised that using a DeprecationWarning given that these are hidden by default. I suggest making a qcodes deprecation warning that inherits from runtime warning

Reply:
Does it make sense to make stacklevel an optional arg with a default value of 2

Reply:
Hehe, I was thinking the same thing, but I guess I was too lazy and thought we could add it once we need it. But you are right, since I am touching this code anyways, I will add it.

Reply:
And good point about the `RuntimeWarning`!